### DISCUSSION TITLE
custom field permission constraints

### DISCUSSION THREAD

**Discussion by romekj82** [netbox-community/netbox#6033] [Q&A]
Hi, 

I have an issue with creating permissions based on a custom field.
I have few device custom fields, and one of them is "lent" where you can choose from the table.
I would like to create permission that gives access to devices that have custom filed "lent" with "TEAM A".
I tried with this, without success:
{"custom_field_data__lent": "TEAM A"},
or
{"custom_field_data__lent__name": "TEAM A"}

Please help.

**Comment by proudbro** [comment]
You should create permission for dcim -> device model 
`{"custom_field_data__lent": "TEAM A"}`
![image](https://user-images.githubusercontent.com/60253505/112178465-d077b600-8c0a-11eb-9c1f-759b73d21c17.png)

**Reply by romekj82** [reply]
yes, i did that, my permission is for dcim- device, but is does not work.  I am not sure if the "custom_field_data__lent" syntax is correct. The internal field name is "lent"
![a](https://user-images.githubusercontent.com/81226484/112217320-c9f53880-8c22-11eb-9d50-4a7ab61c715c.png)

**Reply by romekj82** [reply]
I found the issue. the syntax is good. 
The issue is:
TEAM A has permission to store devices in RACK A
TEAM B has permission to store devices in RACK B
I would like to do something like that:
Team B  lends the device to team A and chooses the TEAM A in  the field "lent"  
And now :
Team A does not have permission change the device, Team B does not have permission to put the device in Team A rack.
How to solve this?
would be great when Team which borrows the equipment should put it in its rack, but when the device is in another rack it does not have permission.

**Reply by romekj82** [reply]
will something like that work: {"rack__tenant__slug" : "tenant__slug"}???
